core/environment/environment.go

```diff
@@ -519,6 +529,34 @@ func (env *Environment) Id() uid.ID {
 	return env.id
 }
 
+func (env *Environment) GetName() string {
+	if env == nil {
+		return ""
+	}
+	env.Mu.RLock()
+	defer env.Mu.RUnlock()
+	return env.name
+}
+
+func (env *Environment) SetName(name string) {
+	if env == nil {
+		return
+	}
+	env.Mu.Lock()
+	defer env.Mu.Unlock()
+	env.name = name
```

Comment:
The interface is correct and well designed, but I think the implementation misses a detail.

I count 2 occurrences of the environment name: the `env.name` member variable, and the `environment_name` `UserVars` entry, which ends up in the common variables stack of the template system for the given environment.

These two must be kept in sync: your `GetName` may only read `env.name`, but your `SetName` must also write into `env.UserVars`, otherwise a subsequent query for the env name from within the environment template context will yield inconsistent results.